Upstream-Status: Inappropriate [OE-specific]
Index: pjproject-2.15.1/configure.ac
===================================================================
--- pjproject-2.15.1.orig/configure.ac
+++ pjproject-2.15.1/configure.ac
@@ -2743,6 +2743,10 @@ AC_ARG_ENABLE(libwebrtc,
                         ac_webrtc_instset=neon
                         ac_webrtc_cflags="-DWEBRTC_ARCH_ARM64"
                     ;;
+		    arm-*)
+			ac_webrtc_instset=neon
+			ac_webrtc_cflags="-DWEBRTC_ARCH_ARMV7"
+		    ;;
                     riscv*)
                         ac_webrtc_instset=generic
                     ;;
@@ -3002,4 +3006,3 @@ Further customizations can be put in:
 
 The next step now is to run 'make dep' and 'make'.
 ])
-
