From a20fff46605599be806908814b231c76023a368b Mon Sep 17 00:00:00 2001
From: Thomas Roos <throos@amazon.de>
Date: Wed, 21 Jan 2026 14:12:53 +0000
Subject: [PATCH] Remove Python version restriction for compatibility with
 Python 3.14+

Upstream-Status: Inappropriate [embedded specific]
---
 setup.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/setup.py b/setup.py
index 004d3eb..8816d4b 100644
--- a/setup.py
+++ b/setup.py
@@ -17,10 +17,6 @@ else:
 
 
 major_version, minor_version = sys.version_info[:2]
-if not (major_version == 3 and 7 <= minor_version <= 13):
-    sys.stderr.write("Sorry, only Python 3.7 - 3.13 are "
-                     "supported at this time.\n")
-    exit(1)
 
 setup(
     name="geojson",
