Upstream-Status: Inappropriate [OE-specific]
diff --git a/configure.ac b/configure.ac
index 6389bdd..22245bc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -41,7 +41,7 @@ AC_SUBST([astlibdir],         ['${libdir}'])dnl
 AC_SUBST([astmoddir],         ['${libdir}/asterisk/modules'])dnl
 AC_SUBST([astmandir],         ['${mandir}'])dnl
 AC_SUBST([astvarlibdir],      ['${localstatedir}/lib/asterisk'])dnl
-AC_SUBST([astdatadir],        ['${astvarlibdir}'])dnl
+AC_SUBST([astdatadir],        ['${datadir}/asterisk'])dnl
 AC_SUBST([astdbdir],          ['${astvarlibdir}'])dnl
 AC_SUBST([astkeydir],         ['${astvarlibdir}'])dnl
 AC_SUBST([astspooldir],       ['${localstatedir}/spool/asterisk'])dnl
